package/src/skia/web/api/JsiSkShaderFactory.ts

```diff
@@ -0,0 +1,178 @@
+    mode: TileMode,
+    localMatrix?: SkMatrix,
+    flags?: number
+  ) {
+    return new JsiSkShader(
+      this.CanvasKit,
+      this.CanvasKit.Shader.MakeLinearGradient(
+        toValue(start),
+        toValue(end),
+        colors,
+        pos,
+        ckEnum(mode),
+        localMatrix === undefined ? undefined : toValue(localMatrix),
+        flags
+      )
+    );
+  }
+
+  MakeRadialGradient(
+    center: SkPoint,
+    radius: number,
+    colors: SkColor[],
+    pos: number[] | null,
+    mode: TileMode,
+    localMatrix?: SkMatrix,
+    flags?: number
+  ) {
+    return new JsiSkShader(
+      this.CanvasKit,
+      this.CanvasKit.Shader.MakeRadialGradient(
+        toValue(center),
+        radius,
+        colors,
+        pos,
+        ckEnum(mode),
+        localMatrix === undefined ? undefined : toValue(localMatrix),
+        flags
+      )
+    );
+  }
+
+  MakeTwoPointConicalGradient(
+    start: SkPoint,
+    startRadius: number,
+    end: SkPoint,
+    endRadius: number,
+    colors: SkColor[],
+    pos: number[] | null,
+    mode: TileMode,
+    localMatrix?: SkMatrix,
+    flags?: number
+  ) {
+    return new JsiSkShader(
+      this.CanvasKit,
+      this.CanvasKit.Shader.MakeTwoPointConicalGradient(
+        toValue(start),
+        startRadius,
+        toValue(end),
+        endRadius,
+        colors,
+        pos,
+        ckEnum(mode),
+        localMatrix === undefined ? undefined : toValue(localMatrix),
+        flags
+      )
+    );
+  }
+
+  MakeSweepGradient(
+    cx: number,
+    cy: number,
+    colors: SkColor[],
+    pos: number[] | null,
+    mode: TileMode,
+    localMatrix?: SkMatrix | null,
+    flags?: number,
+    startAngleInDegrees?: number,
+    endAngleInDegrees?: number
+  ) {
+    return new JsiSkShader(
+      this.CanvasKit,
+      this.CanvasKit.Shader.MakeSweepGradient(
+        cx,
+        cy,
+        colors,
+        pos,
+        ckEnum(mode),
+        localMatrix === undefined || localMatrix === null
+          ? undefined
+          : toValue(localMatrix),
+        flags,
+        startAngleInDegrees,
+        endAngleInDegrees
+      )
+    );
+  }
+
+  MakeTurbulence(
+    baseFreqX: number,
+    baseFreqY: number,
+    octaves: number,
+    seed: number,
+    tileW: number,
+    tileH: number
+  ) {
+    return new JsiSkShader(
+      this.CanvasKit,
+      this.CanvasKit.Shader.MakeTurbulence(
+        baseFreqX,
+        baseFreqY,
+        octaves,
+        seed,
+        tileW,
+        tileH
+      )
+    );
+  }
+
+  MakeFractalNoise(
+    baseFreqX: number,
+    baseFreqY: number,
+    octaves: number,
+    seed: number,
+    tileW: number,
+    tileH: number
+  ) {
+    return new JsiSkShader(
+      this.CanvasKit,
+      this.CanvasKit.Shader.MakeFractalNoise(
+        baseFreqX,
+        baseFreqY,
+        octaves,
+        seed,
+        tileW,
+        tileH
+      )
+    );
+  }
+
+  MakeBlend(mode: BlendMode, one: SkShader, two: SkShader) {
```

Comment:
Should the pattern where we test for null be implemented here as well?

Reply:
This cannot return null?